### ISSUE TITLE
[Platform Compat Analyzer] Attributes are not respected on property getters/setters

### ISSUE THREAD

**Issue by jeffhandley** [dotnet/roslyn-analyzers#4071]
### Describe the bug

Calling property getters or setters that are annotated with `[SupportedOSPlatform]` or `[UnsupportedOSPlatform]` does not raise diagnostics. When the attribute is applied to the property itself, diagnostics are raised as expected.

### Steps To Reproduce

```csharp
        [SupportedOSPlatform("windows")]
        public static bool WindowsOnlyProperty { get; set; } // References to this property DO report diagnostics as expected

        public static bool WindowsOnlyPropertyGetter
        {
            [SupportedOSPlatform("windows")] // References to this property getter DO NOT report diagnostics as expected
            get { return true; }
            set { }
        }

        public static bool WindowsOnlyPropertySetter
        {
            get { return true; }

            [SupportedOSPlatform("windows")] // References to this property setter DO NOT report diagnostics as expected
            set { }
        }
```
### Expected behavior

Annotations on property getters and setters should be respected and should result in diagnostics when referenced in code that is reachable in unsupported platforms.

### Actual behavior

No diagnostic is raised when referencing property getters or setters that are reachable with unsupported platforms.

### Analyzer

**Platform Compatibility Analyzer** (local RC1 build)

**Diagnostic ID**: [CA1416](https://docs.microsoft.com/visualstudio/code-quality/ca1416)

## Additional context

Some properties on [`System.Console`](https://github.com/dotnet/runtime/blob/master/src/libraries/System.Console/src/System/Console.cs) have been annotated using this approach of applying the attribute to property setters, while leaving the property getters supported on all platforms.  This bug was discovered when trying to access the `Console.WindowHeight` property setter and the `Console.CursorVisible` property getter with code reachable in platforms other than Windows.

We should attempt to fix this in RC1 with the introduction of the analyzer.

/cc @jmarolf

**Comment by terrajobst** [comment]
Oh... I haven't seen this! Yeah, this needs to be fixed for RTM (unlike #4072). We applied a ton of attributes to just getters/setters.
